File: com.systemincloud.examples.tasks.javatask/src/test/java/tasks/hold/Default.java
package tasks.hold;

import com.systemincloud.modeler.tasks.javatask.api.InputPort;
import com.systemincloud.modeler.tasks.javatask.api.JavaTask;
import com.systemincloud.modeler.tasks.javatask.api.OutputPort;
import com.systemincloud.modeler.tasks.javatask.api.annotations.InputPortInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.JavaTaskInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.OutputPortInfo;
import com.systemincloud.modeler.tasks.javatask.api.data.Text;

@JavaTaskInfo
public class Default extends JavaTask {

	@InputPortInfo(name = "In 1", dataType = Text.class, hold = true)
	public InputPort in1;
	@InputPortInfo(name = "In 2", dataType = Text.class)
	public InputPort in2;
	@OutputPortInfo(name = "Out", dataType = Text.class)
	public OutputPort out;

	@Override
	public void execute(int grp) {
		out.putData(new Text(in1.getData(Text.class).getValue() + in2.getData(Text.class).getValue()));
	}

}

File: com.systemincloud.examples.tasks.pythontask/src/test/py/tasks/hold/Default.py
from sicpythontask.PythonTaskInfo import PythonTaskInfo
from sicpythontask.PythonTask import PythonTask
from sicpythontask.InputPort import InputPort
from sicpythontask.OutputPort import OutputPort
from sicpythontask.data.Text import Text

@PythonTaskInfo
class Default(PythonTask):

    def __init_ports__(self):
        self.in1 = InputPort(name="in1", data_type=Text)
        self.in2 = InputPort(name="in2", data_type=Text, hold=True)
        self.out = OutputPort(name="out", data_type=Text)

    def execute(self, grp):
        self.out.put_data(Text(self.in1.get_data(Text).values[0] + self.in2.get_data(Text).values[0]));
